ignite/contrib/engines.py

```diff
@@ -36,10 +37,12 @@ def _detach_hidden(hidden):
     Auxillary function to cut the backpropagation graph by detaching the hidden
     vector.
     """
-    if isinstance(hidden, collections.Iterable):
+    if isinstance(hidden, torch.Tensor):
+        return hidden.detach()
+    elif isinstance(hidden, tuple):
```

Comment:
nit: `elif isinstance(hidden, tuple):` -> `elif isinstance(hidden, (list, tuple)):`

Reply:
I removed `collections.Iterable` because it turns out that `Tensor` implements that (in an unexpected way).

In Pytorch, the hidden vectors for recurrent models are either a `Tensor` or a `tuple` of `Tensor`. I'll mimic the function you mentioned so that this can generalize to user-defined structures if needed.